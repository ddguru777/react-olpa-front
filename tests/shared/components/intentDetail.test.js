/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import renderer from "react-test-renderer";
import IntentDetail from "shared/components/intentDetail";

describe("components/intentDetail", () => {
  it("renders correctly", () => {
    const intent = {topic:"some intent"};
    const component = renderer.create(<IntentDetail intent={intent} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
