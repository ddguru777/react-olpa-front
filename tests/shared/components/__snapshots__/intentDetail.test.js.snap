// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/intentDetail renders correctly 1`] = `
<div
  className="mrb-action-panel list-box"
>
  <div
    className="list-content"
  >
    <div
      className="mrb-sublist"
    >
      <div
        className="mrb-subheader"
      >
        <i
          className="mrb-subheader-right material-icons"
          disabled={false}
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="0"
        >
          keyboard_arrow_up
        </i>
        <div>
          input
        </div>
      </div>
      <div
        style={
          Object {
            "display": "block",
          }
        }
      >
        <ul
          className="mdc-list"
        />
        <li
          className="selectableListItem onFocusAction mdl-list_action mdc-list-item"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="menuitem"
          style={
            Object {
              "color": "rgb(213, 0, 0)",
              "padding": "0px 16px",
            }
          }
        >
          <i
            aria-hidden="true"
            className="mdc-list-item__graphic material-icons"
          >
            format_quote
          </i>
          Add an input sentence
        </li>
      </div>
    </div>
    <hr />
    <div
      className="mrb-sublist"
    >
      <div
        className="mrb-subheader"
      >
        <i
          className="mrb-subheader-right material-icons"
          disabled={false}
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="0"
        >
          keyboard_arrow_up
        </i>
        <div>
          output
        </div>
      </div>
      <div
        style={
          Object {
            "display": "block",
          }
        }
      >
        <ul
          className="mdc-list"
        />
        <li
          className="selectableListItem onFocusAction mdl-list_action mdc-list-item"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="menuitem"
          style={
            Object {
              "color": "rgb(213, 0, 0)",
              "padding": "0px 16px",
            }
          }
        >
          <i
            aria-hidden="true"
            className="mdc-list-item__graphic material-icons"
          >
            chat_bubble_outline
          </i>
          Add an output response
        </li>
      </div>
    </div>
    <hr />
    <div
      className="mrb-sublist"
    >
      <div
        className="mrb-subheader"
      >
        <i
          className="mrb-subheader-right material-icons"
          disabled={false}
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="0"
        >
          keyboard_arrow_down
        </i>
        <div>
          Parameters
        </div>
      </div>
      <div
        style={
          Object {
            "display": "none",
          }
        }
      >
        <ul
          className="mdc-list"
        >
          <li
            className="mdc-list-item"
            onKeyPress={[Function]}
            role="menuitem"
          >
            <i
              aria-hidden="true"
              className="mdc-list-item__graphic material-icons"
            >
              title
            </i>
            Topic
            <span
              className="mdc-list-item__meta"
              onClick={null}
              onKeyUp={[Function]}
              role="button"
              tabIndex="0"
            >
              <button
                className="mrb-action-button mdc-button"
                disabled={false}
                id={null}
                onClick={[Function]}
                type="button"
              >
                
                some intent
              </button>
            </span>
          </li>
          <li
            className="mdc-list-item"
            onKeyPress={[Function]}
            role="menuitem"
          >
            <i
              aria-hidden="true"
              className="mdc-list-item__graphic material-icons"
            >
              link
            </i>
            Previous
            <span
              className="mdc-list-item__meta"
              onClick={null}
              onKeyUp={[Function]}
              role="button"
              tabIndex="0"
            >
              <button
                className="mrb-action-button mdc-button"
                disabled={false}
                id={null}
                onClick={[Function]}
                type="button"
              >
                
                NONE
              </button>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
